src/rlookup.c

```diff
@@ -303,6 +301,31 @@ static RSValue *jsonValToValue(RedisModuleCtx *ctx, RedisJSON json) {
   return NULL;
 }
 
+// Get the value from an iterator and free the iterator
+// Return 1, and set rsv to the value, if value exists
+// Return 0 otherwise
+static int jsonIterToValue(RedisModuleCtx *ctx, JSONResultsIterator iter, RSValue **rsv) {
+
+  int res = 1;
+  if (japi->len(iter) == 1) {
+    RedisJSON jsonValue = japi->next(iter);
+    if (jsonValue) {
+      *rsv = jsonValToValue(ctx, jsonValue);
+    } else {
+      res = 0;
```

Comment:
start with `res = 0` and set to `1` if successful?

Reply:
Assuming the positive scenario to be more common, I took the opposite direction.

Reply:
We often start with REDISMODULE_ERR and set it to RM_OK. It saves a branch and is easier on the eye.
Not crucial :)